### PR TITLE
DN5381 - feature: ioc features

### PR DESCRIPTION
**Pull request by mjbradford89**

### CODE REVIEW

File: src/polyswarm/client/search.py [hunk 1]
```diff
@@ -71,6 +71,74 @@ def metadata(ctx, query_string, include, exclude, ip, url, domain):
     for metadata_result in api.search_by_metadata(query_string, include=include, exclude=exclude, ips=ip, urls=url, domains=domain):
         output.metadata(metadata_result)
 
+@search.command('iocs_by_hash', short_help='Retrieve the IOCs associated with the artifact hash.')
```
Comment (sbneto) [inline src/polyswarm/client/search.py:74]: I think we are using dashes instead of underscores for the multi-word commands.

**Review by sbneto — COMMENTED** [review]
Just one adjstment. Maybe @mrtizmoatwork wants to comment on the command name. Also, if you add docstrings to click command functions they turn into the description given when you do `--help` with the `cli`.

**Comment by mjbradford89** [comment]
> regarding the name of the commands, can you show me example command lines for each new command?

```bash
$ polyswarm -a $KEY search iocs-by-hash sha256 18e5b8fe65e8f73c3a4a637c258c02aeec8a6ab702b15b7ee73f5631a9879e40
============================= IOCs =============================
ImpHash:
IPs: 192.0.0.1, 2.2.2.2
URLs: polyswarm.io
TTPs: TTP123, TTP456

$ polyswarm -a $KEY search artifact-by-ioc ip 1.2.3.4
============================= IOCs =============================
SHA256: 18e5b8fe65e8f73c3a4a637c258c02aeec8a6ab702b15b7ee73f5631a9879e40
SHA256: 275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0a

$ polyswarm -a $KEY -u $URL search check-known-hosts --ip 2.2.2.2 --domain google.com
============================= Known IOC =============================
ID: 1
type: ip
host: 2.2.2.2
source: xyz list
good: True

$ polyswarm -a $KEY -u $URL search add-known-good-host ip 2.2.2.2 "xyz list"
============================= Known IOC =============================
ID: 1
type: ip
host: 2.2.2.2
source: xyz list
good: True

$ polyswarm -a $KEY -u $URL search update-known-good-host 2 domain polyswarm.io "abc list" 
============================= Known IOC =============================
ID: 2
type: domain
host: polyswarm.io
source: abc list
good: True

```

**Comment by mrtizmoatwork** [comment]
ok. I'd like to clean up the commands a little. Can we do something like this:

`
$ polyswarm -a $KEY search ioc sha256 18e5b8fe65e8f73c3a4a637c258c02aeec8a6ab702b15b7ee73f5631a9879e40

$ polyswarm -a $KEY search ioc ip 1.2.3.4

$ polyswarm -a $KEY -u $URL search known --ip 2.2.2.2 --domain google.com

$ polyswarm -a $KEY -u $URL known add --good ip 2.2.2.2 "xyz list"
$ polyswarm -a $KEY -u $URL known add --bad ip 2.2.2.2 "xyz list"

$ polyswarm -a $KEY -u $URL known update --id 2 --good  domain polyswarm.io "abc list" 

$ polyswarm -a $KEY -u $URL known delete --id 2


`

**Comment by mrtizmoatwork** [comment]
@sbneto any other feedback on my suggested command syntax there?

**Comment by mjbradford89** [comment]
Updated to match the suggested syntax with minor difference of no `bad` flag.  instead with click it's easier to use one argument and pass a bool, i.e.
```bash
--good False
```

**Review by admin-sbneto — APPROVED** [review]

**Pull request merged**